frontend/src/ManageGroupPermissions/ManageGroupPermissions.jsx

```diff
@@ -42,24 +50,130 @@ class ManageGroupPermissionsComponent extends React.Component {
+          this.setState({
+            newGroupName: '',
+            creatingGroup: false,
+            selectedGroupPermissionId: data?.id,
+            selectedGroup: data?.group,
+            isDuplicatingGroup: false,
+            showDuplicateGroupModal: false,
+            groupDuplicateOption: { addPermission: true, addApps: true, addUsers: true },
+          });
+        });
+
+        toast.success('Group duplicated successfully!');
+      })
+      .catch((err) => {
+        this.setState({
+          isDuplicatingGroup: false,
+          groupDuplicateOption: { addPermission: true, addApps: true, addUsers: true },
+        });
+        console.error('Error occured in duplicating: ', err);
+        toast.error('Could not duplicate group.\nPlease try again!');
+      });
+  };
+
+  toggleShowDuplicateModal = () => {
+    this.setState((prevState) => ({
+      showDuplicateGroupModal: !prevState.showDuplicateGroupModal,
+      groupToDuplicate: '',
+      groupDuplicateOption: { addPermission: true, addApps: true, addUsers: true },
+    }));
+  };
+
+  renderPopoverContent = (props, compoParam) => {
+    const { groupName, id } = compoParam;
+    const deleteGroup = () => {
+      this.deleteGroup(id);
+    };
+
+    const duplicateGroup = () => {
+      this.showDuplicateDiologBox(id);
+    };
+
+    return (
+      <div
+        {...props}
+        style={{
+          position: 'absolute',
+          ...props.style,
+        }}
+      >
+        <Popover
+          id="popover-group-menu"
+          className={this.props.darkMode ? 'popover-group-menu dark-theme' : 'popover-group-menu'}
+          placement="bottom"
+        >
+          <Popover.Body bsPrefix="popover-body">
+            <div>
+              <Field
+                leftIcon="copy"
+                leftIconWidth="20"
+                leftViewBox="0  0 20 20"
+                text={'Duplicate group'}
+                // text={t('header.organization.manageGroups.groupOptions.deleteGroup', 'Duplicate Group')}
+                onClick={duplicateGroup}
+              />
+              <Field
+                leftIcon="delete"
+                leftIconWidth="18"
+                leftIconHeight="18"
+                leftViewBox="0  0 20 20"
+                text={'Delete group'}
+                tooltipId="tooltip-for-delete"
+                tooltipContent="Cannot delete default group"
+                onClick={groupName == 'all_users' || groupName == 'admin' ? {} : deleteGroup}
+                buttonDisable={groupName == 'all_users' || groupName == 'admin'}
+              />
+            </div>
+          </Popover.Body>
+        </Popover>
+        {(groupName == 'all_users' || groupName == 'admin') && (
+          <Tooltip
+            id="tooltip-for-delete"
+            className="tooltip"
+            place="left"
+            style={{
+              zIndex: 99999,
+            }}
+            show={groupName == 'all_users' || groupName == 'admin'}
```

Comment:
I can see the same condition more than 2 times on this file. shall we move this to a variable and reuse it? 
```suggestion
            const isDefaultGroup = groupName == 'all_users' || groupName == 'admin';
            show={isDefaultGroup}
```